server/datastore/mysql/apple_mdm.go

```diff
@@ -3465,15 +3465,17 @@ WHERE
 	return &profile, nil
 }
 
-func bulkDeleteHostDiskEncryptionKeysDB(ctx context.Context, tx sqlx.ExtContext, hostIDs []uint) error {
+func bulkDeleteMacosHostDiskEncryptionKeysDB(ctx context.Context, tx sqlx.ExtContext, hostIDs []uint) error {
 	if len(hostIDs) == 0 {
 		return nil
 	}
 
-	query, args, err := sqlx.In(
-		"DELETE FROM host_disk_encryption_keys WHERE host_id IN (?)",
-		hostIDs,
-	)
+	query, args, err := sqlx.In(`
+		DELETE hdek 
+		FROM host_disk_encryption_keys hdek
+		INNER JOIN hosts h ON hdek.host_id = h.id 
+		WHERE h.platform = 'darwin' AND hdek.host_id IN (?)`,
+		hostIDs)
```

Comment:
scoping this deletion process to macos only, as moving a host to a new team on linux should not delete linux escrowed keys.

Reply:
Seems like we may want this behavior regardless of OS, but this fn should've been moved so it's more obvious that it's cross-platform? See context in https://github.com/fleetdm/fleet/commit/a59b8a5096c8cadde259a69b5038415ab0ee049c.

Though this probably needs more intelligence to only un-escrow when moving from a team that requires disk encryption to one that doesn't?

Since right now moving teams for a BitLocker host would also delete the key, right?

Reply:
reverted this change for now to unblock the change.  Waiting to hear back from product on desired key cleanup behavior

Reply:
We'll want to warn the user before allowing the transfer to avoid "data loss".
E.g. device returned to admin, admin transfers host to a `Maintenance` team (or something else, or maybe accidental transfer), escrow key gone...